# MCO1 

// Project Details Details

### GARYNATION (S13)
1. Bien Aaron Miranda
2. Luis Miguel Rana
3. Karl Andre Aquino
4. Dominic Luis Baccay

### Python Libraries

Import numpy and pandas.

pandas is a software library for Python which provides data structures and data analysis tools.

In [731]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns


## The Dataset

    /////Give details about the dataset, what is inside the dataset, the variables, etc.
    
    A collection of YouTube giants, this dataset offers a perfect avenue to analyze and gain valuable insights from the luminaries of the platform. With comprehensive details on top creators' subscriber counts, video views, upload frequency, country of origin, earnings, and more, this treasure trove of information is a must-explore for aspiring content creators, data enthusiasts, and anyone intrigued by the ever-evolving online content landscape.

In [732]:
#encoding came from the discussion so that it is readable
youtube_df = pd.read_csv("Global YouTube Statistics.csv", encoding='iso-8859-1')

If you view the `.csv` file in Excel, you can see that our dataset contains about 995 **observations** (rows) across 28 **variables** (columns). The following are the descriptions of each variable in the dataset.

- **`rank`**: Position of the YouTube channel based on the number of subscribers
- **`Youtuber`**: Name of the YouTube channel
- **`subscribers`**: Number of subscribers to the channel
- **`video views`**: Overall count views to the channel
- **`category`**: Category or niche of the channel
- **`Title`**: Title
- **`uploads`**: Total number of videos uploaded on the channel
- **`Country`**: Country where the YouTube channel originates
- **`Abbreviation`**: Abbreviation of the country
- **`channel_type`**: Type of the YouTube channel (e.g., individual, brand)
- **`video_views_rank`**: Ranking of the channel based on total video views
- **`country_rank`**: Ranking of the channel based on the number of subscribers within its country
- **`channel_type_rank`**: Ranking of the channel based on its type (individual or brand)
- **`video_views_for_the_last_30_days`**: Total video views in the last 30 days
- **`lowest_monthly_earnings`**: Lowest estimated monthly earnings from the channel \$
- **`highest_monthly_earnings`**: Highest estimated monthly earnings from the channel \$
- **`lowest_yearly_earnings`**: Lowest estimated yearly earnings from the channel \$
- **`highest_yearly_earnings`**: Highest estimated yearly earnings from the channel $
- **`subscribers_for_last_30_days`**: Number of new subscribers gained in the last 30 days
- **`created_year`**: Year when the YouTube channel was created
- **`created_month`**: Month when the YouTube channel was created
- **`created_date`**: Exact date of the YouTube channel's creation
- **`Gross tertiary education enrollment (\%)`**: Percentage of the population enrolled in tertiary education in the country %
- **`Population`**: Total population of the country
- **`Unemployment rate`**: Unemployment rate in the country %
- **`Urban_population`**: Percentage of the population living in urban areas %
- **`Latitude`**: Latitude coordinate of the country's location
- **`Longitude`**: Longitude coordinate of the country's location

## Reading the Dataset

Our first step is to load the dataset using pandas. This will load the dataset into a pandas DataFrame. To load the dataset, we use the read_csv function. Note that you may need to change the path depending on the location of the file in your machine.

Will also tackle about what can be seen in the Dataset as a DataFrame

In [733]:
youtube_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 995 entries, 0 to 994
Data columns (total 28 columns):
 #   Column                                   Non-Null Count  Dtype  
---  ------                                   --------------  -----  
 0   rank                                     995 non-null    int64  
 1   Youtuber                                 995 non-null    object 
 2   subscribers                              995 non-null    int64  
 3   video views                              995 non-null    float64
 4   category                                 949 non-null    object 
 5   Title                                    995 non-null    object 
 6   uploads                                  995 non-null    int64  
 7   Country                                  873 non-null    object 
 8   Abbreviation                             873 non-null    object 
 9   channel_type                             965 non-null    object 
 10  video_views_rank                         994 non-n

### Examining the observations within the Dataset

    
To gain a better understanding of the observations within the dataset, it's essential to examine their appearance or visual representation. This can provide valuable insights and help you make informed decisions based on the data.

In [734]:
youtube_df.head(10)

rank                    Youtuber  subscribers   video views  \
0     1                    T-Series    245000000  2.280000e+11   
1     2              YouTube Movies    170000000  0.000000e+00   
2     3                     MrBeast    166000000  2.836884e+10   
3     4  Cocomelon - Nursery Rhymes    162000000  1.640000e+11   
4     5                   SET India    159000000  1.480000e+11   
5     6                       Music    119000000  0.000000e+00   
6     7         ýýý Kids Diana Show    112000000  9.324704e+10   
7     8                   PewDiePie    111000000  2.905804e+10   
8     9                 Like Nastya    106000000  9.047906e+10   
9    10               Vlad and Niki     98900000  7.718017e+10   

           category                       Title  uploads        Country  \
0             Music                    T-Series    20082          India   
1  Film & Animation               youtubemovies        1  United States   
2     Entertainment                     MrBeast      741  United States   
3         Education  Cocomelon - Nursery Rhymes      966  United States   
4             Shows                   SET India   116536          India   
5               NaN                       Music        0            NaN   
6    People & Blogs         ýýý Kids Diana Show     1111  United States   
7            Gaming                   PewDiePie     4716          Japan   
8    People & Blogs            Like Nastya Vlog      493         Russia   
9     Entertainment               Vlad and Niki      574  United States   

  Abbreviation   channel_type  ...  subscribers_for_last_30_days  \
0           IN          Music  ...                     2000000.0   
1           US          Games  ...                           NaN   
2           US  Entertainment  ...                     8000000.0   
3           US      Education  ...                     1000000.0   
4           IN  Entertainment  ...                     1000000.0   
5          NaN          Music  ...                           NaN   
6           US  Entertainment  ...                           NaN   
7           JP  Entertainment  ...                           NaN   
8           RU         People  ...                      100000.0   
9           US  Entertainment  ...                      600000.0   

   created_year  created_month  created_date  \
0        2006.0            Mar          13.0   
1        2006.0            Mar           5.0   
2        2012.0            Feb          20.0   
3        2006.0            Sep           1.0   
4        2006.0            Sep          20.0   
5        2013.0            Sep          24.0   
6        2015.0            May          12.0   
7        2010.0            Apr          29.0   
8        2016.0            Jan          14.0   
9        2018.0            Apr          23.0   

   Gross tertiary education enrollment (%)    Population  Unemployment rate  \
0                                     28.1  1.366418e+09               5.36   
1                                     88.2  3.282395e+08              14.70   
2                                     88.2  3.282395e+08              14.70   
3                                     88.2  3.282395e+08              14.70   
4                                     28.1  1.366418e+09               5.36   
5                                      NaN           NaN                NaN   
6                                     88.2  3.282395e+08              14.70   
7                                     63.2  1.262266e+08               2.29   
8                                     81.9  1.443735e+08               4.59   
9                                     88.2  3.282395e+08              14.70   

   Urban_population   Latitude   Longitude  
0       471031528.0  20.593684   78.962880  
1       270663028.0  37.090240  -95.712891  
2       270663028.0  37.090240  -95.712891  
3       270663028.0  37.090240  -95.712891  
4       471031528.0  20.593684   78.962880  
5               NaN        NaN  

# Cleaning the Dataset

This process aims to enhance the quality and reliability of the dataset.

### Before Cleaning:
Number of Observations: 995<br>
Number of Columns: 28<br>
Data types: float64(18), int64(3), object(7)<br>

#### Initial Observations: 
- There are special characters in the 'Youtuber' column.
- There are missing/null values in the dataset.
- Some variables are similar, therefore, unnecessary.
- Some data types are inappropriate for their respective variables.

#### Variables to be Observed:
We will be dropping all columns/variables except the following.
- **`Youtuber`**: Name of the YouTube channel
- **`subscribers`**: Number of subscribers to the channel
- **`video views`**: Overall count views to the channel
- **`category`**: Category or niche of the channel
- **`uploads`**: Total number of videos uploaded on the channel
- **`Country`**: Country where the YouTube channel originates
- **`channel_type`**: Type of the YouTube channel (e.g., individual, brand)
- **`lowest_monthly_earnings`**: Lowest estimated monthly earnings from the channel \$
- **`highest_monthly_earnings`**: Highest estimated monthly earnings from the channel \$
- **`lowest_yearly_earnings`**: Lowest estimated yearly earnings from the channel \$
- **`highest_yearly_earnings`**: Highest estimated yearly earnings from the channel $
- **`created_year`**: Year when the YouTube channel was created
- **`Population`**: Total population of the country


In [735]:
# Declare the variables that will be used
columns_to_keep = ['Youtuber', 'subscribers', 'video views', 'category', 'channel_type', 'uploads', 'Country', 'lowest_monthly_earnings', 'highest_monthly_earnings', 'lowest_yearly_earnings', 'highest_yearly_earnings', 'created_year', 'Population']
columns_to_drop = list(set(youtube_df.columns) - set(columns_to_keep))

# Drop unused columns (variables)
youtube_df.drop(columns_to_drop, axis=1, inplace=True)

youtube_df

Youtuber  subscribers   video views          category  \
0                      T-Series    245000000  2.280000e+11             Music   
1                YouTube Movies    170000000  0.000000e+00  Film & Animation   
2                       MrBeast    166000000  2.836884e+10     Entertainment   
3    Cocomelon - Nursery Rhymes    162000000  1.640000e+11         Education   
4                     SET India    159000000  1.480000e+11             Shows   
..                          ...          ...           ...               ...   
990               Natan por Aï¿     12300000  9.029610e+09            Sports   
991    Free Fire India Official     12300000  1.674410e+09    People & Blogs   
992                       Panda     12300000  2.214684e+09               NaN   
993                 RobTopGames     12300000  3.741235e+08            Gaming   
994                Make Joke Of     12300000  2.129774e+09            Comedy   

     uploads         Country   channel_type  lowest_monthly_earnings  \
0      20082           India          Music                 564600.0   
1          1   United States          Games                      0.0   
2        741   United States  Entertainment                 337000.0   
3        966   United States      Education                 493800.0   
4     116536           India  Entertainment                 455900.0   
..       ...             ...            ...                      ...   
990     1200          Brazil  Entertainment                 138100.0   
991     1500           India          Games                  16200.0   
992     2452  United Kingdom          Games                     17.0   
993       39          Sweden          Games                    968.0   
994       62           India         Comedy                   6000.0   

     highest_monthly_earnings  lowest_yearly_earnings  \
0                  9000000.00              6800000.00   
1                        0.05                    0.04   
2                  5400000.00              4000000.00   
3                  7900000.00              5900000.00   
4                  7300000.00              5500000.00   
..                        ...                     ...   
990                2200000.00              1700000.00   
991                 258900.00               194200.00   
992                    268.00                  201.00   
993                  15500.00                11600.00   
994                  96000.00                72000.00   

     highest_yearly_earnings  created_year    Population  
0               1.084000e+08        2006.0  1.366418e+09  
1               5.800000e-01        2006.0  3.282395e+08  
2               6.470000e+07        2012.0  3.282395e+08  
3               9.480000e+07        2006.0  3.282395e+08  
4               8.750000e+07        2006.0  1.366418e+09  
..                       ...           ...           ...  
990             2.650000e+07        2017.0  2.125594e+08  
991             3.100000e+06        2018.0  1.366418e+09  
992             3.200000e+03        2006.0  6.683440e+07  
993             1.858000e+05        2012.0  1.028545e+07  
994             1.200000e+06        2017.0  1.366418e+09  

[995 rows x 13 columns]

### The "Youtuber" variable - Names of Youtubers

     We want to check if there are any observations that has no "Youtuber" values, because if there are no values present, we will be excluding these observations from the list.
     

In [736]:
# Check for null values
nullyoutuber_df = youtube_df[youtube_df["Youtuber"].isnull()]
nullyoutuber_df

Empty DataFrame
Columns: [Youtuber, subscribers, video views, category, uploads, Country, channel_type, lowest_monthly_earnings, highest_monthly_earnings, lowest_yearly_earnings, highest_yearly_earnings, created_year, Population]
Index: []

     We want to check if there are any duplicate "Youtuber" values, if any, the row will be removed.
     

In [737]:
# Check for duplicates
column_name = "Youtuber"
unique_count = youtube_df[column_name].nunique()
observation_count = youtube_df[column_name].count()

print(f"Number of observations in '{column_name}': {observation_count}")
print(f"Number of unique values in '{column_name}': {unique_count}")

Number of observations in 'Youtuber': 995
Number of unique values in 'Youtuber': 995


### The "video views" and "upload" variables

     We want to check if there are any observations that has zero (0)  "video views" or "uploads" as values since we will be excluding these observations from the list.
     - We will not be replacing the "null" (0) values with any measure of central tendency as this may skew our results too much.
     - The removal of these rows keeps our number of observations at approximately 95.37% (949 of 995) of the original count.

In [738]:
# Drop the rows where video views == 0
youtube_df = youtube_df[youtube_df["video views"] != 0]

# Drop the rows where uploads == 0
youtube_df = youtube_df[youtube_df["uploads"] != 0]

youtube_df

Youtuber  subscribers   video views        category  \
0                      T-Series    245000000  2.280000e+11           Music   
2                       MrBeast    166000000  2.836884e+10   Entertainment   
3    Cocomelon - Nursery Rhymes    162000000  1.640000e+11       Education   
4                     SET India    159000000  1.480000e+11           Shows   
6           ýýý Kids Diana Show    112000000  9.324704e+10  People & Blogs   
..                          ...          ...           ...             ...   
990               Natan por Aï¿     12300000  9.029610e+09          Sports   
991    Free Fire India Official     12300000  1.674410e+09  People & Blogs   
992                       Panda     12300000  2.214684e+09             NaN   
993                 RobTopGames     12300000  3.741235e+08          Gaming   
994                Make Joke Of     12300000  2.129774e+09          Comedy   

     uploads         Country   channel_type  lowest_monthly_earnings  \
0      20082           India          Music                 564600.0   
2        741   United States  Entertainment                 337000.0   
3        966   United States      Education                 493800.0   
4     116536           India  Entertainment                 455900.0   
6       1111   United States  Entertainment                 182900.0   
..       ...             ...            ...                      ...   
990     1200          Brazil  Entertainment                 138100.0   
991     1500           India          Games                  16200.0   
992     2452  United Kingdom          Games                     17.0   
993       39          Sweden          Games                    968.0   
994       62           India         Comedy                   6000.0   

     highest_monthly_earnings  lowest_yearly_earnings  \
0                   9000000.0               6800000.0   
2                   5400000.0               4000000.0   
3                   7900000.0               5900000.0   
4                   7300000.0               5500000.0   
6                   2900000.0               2200000.0   
..                        ...                     ...   
990                 2200000.0               1700000.0   
991                  258900.0                194200.0   
992                     268.0                   201.0   
993                   15500.0                 11600.0   
994                   96000.0                 72000.0   

     highest_yearly_earnings  created_year    Population  
0                108400000.0        2006.0  1.366418e+09  
2                 64700000.0        2012.0  3.282395e+08  
3                 94800000.0        2006.0  3.282395e+08  
4                 87500000.0        2006.0  1.366418e+09  
6                 35100000.0        2015.0  3.282395e+08  
..                       ...           ...           ...  
990               26500000.0        2017.0  2.125594e+08  
991                3100000.0        2018.0  1.366418e+09  
992                   3200.0        2006.0  6.683440e+07  
993                 185800.0        2012.0  1.028545e+07  
994                1200000.0        2017.0  1.366418e+09  

[949 rows x 13 columns]

     After removing the observations, we will be resetting the indexes ensure there are no gaps and the data remains continuous.

In [739]:
# Reset the index after dropping rows
youtube_df.reset_index(drop=True, inplace=True)
youtube_df

Youtuber  subscribers   video views        category  \
0                      T-Series    245000000  2.280000e+11           Music   
1                       MrBeast    166000000  2.836884e+10   Entertainment   
2    Cocomelon - Nursery Rhymes    162000000  1.640000e+11       Education   
3                     SET India    159000000  1.480000e+11           Shows   
4           ýýý Kids Diana Show    112000000  9.324704e+10  People & Blogs   
..                          ...          ...           ...             ...   
944               Natan por Aï¿     12300000  9.029610e+09          Sports   
945    Free Fire India Official     12300000  1.674410e+09  People & Blogs   
946                       Panda     12300000  2.214684e+09             NaN   
947                 RobTopGames     12300000  3.741235e+08          Gaming   
948                Make Joke Of     12300000  2.129774e+09          Comedy   

     uploads         Country   channel_type  lowest_monthly_earnings  \
0      20082           India          Music                 564600.0   
1        741   United States  Entertainment                 337000.0   
2        966   United States      Education                 493800.0   
3     116536           India  Entertainment                 455900.0   
4       1111   United States  Entertainment                 182900.0   
..       ...             ...            ...                      ...   
944     1200          Brazil  Entertainment                 138100.0   
945     1500           India          Games                  16200.0   
946     2452  United Kingdom          Games                     17.0   
947       39          Sweden          Games                    968.0   
948       62           India         Comedy                   6000.0   

     highest_monthly_earnings  lowest_yearly_earnings  \
0                   9000000.0               6800000.0   
1                   5400000.0               4000000.0   
2                   7900000.0               5900000.0   
3                   7300000.0               5500000.0   
4                   2900000.0               2200000.0   
..                        ...                     ...   
944                 2200000.0               1700000.0   
945                  258900.0                194200.0   
946                     268.0                   201.0   
947                   15500.0                 11600.0   
948                   96000.0                 72000.0   

     highest_yearly_earnings  created_year    Population  
0                108400000.0        2006.0  1.366418e+09  
1                 64700000.0        2012.0  3.282395e+08  
2                 94800000.0        2006.0  3.282395e+08  
3                 87500000.0        2006.0  1.366418e+09  
4                 35100000.0        2015.0  3.282395e+08  
..                       ...           ...           ...  
944               26500000.0        2017.0  2.125594e+08  
945                3100000.0        2018.0  1.366418e+09  
946                   3200.0        2006.0  6.683440e+07  
947                 185800.0        2012.0  1.028545e+07  
948                1200000.0        2017.0  1.366418e+09  

[949 rows x 13 columns]

### The "category" variable

     We want to check if there are any observations that has null values for the "category" column as we will be replacing their values.

In [740]:
# Check for null values in category
nullcategory_df = youtube_df[youtube_df["category"].isnull()]
nullcategory_df

Youtuber  subscribers   video views  \
85                               Vlad vï¿½ï¿½ï     37900000  2.351015e+10   
124                              BETER B��     32700000  1.918004e+10   
146                         Diana and Roma ESP     30700000  1.235599e+10   
152  ýýýýýýýýýýýýýýý ýýýýýý ýýýýýýýýýýýýýýýýýý     30400000  1.799996e+10   
201                                 D Billions     26200000  3.197746e+10   
239                            shfa show India     24200000  1.572416e+10   
240   Super JoJo - Nursery Rhymes & Kids Songs     24100000  3.297749e+08   
245                         Diana and Roma ARA     24100000  1.104126e+10   
294                       Vlad and Niki Arabic     22500000  1.061370e+10   
301                          Vlad and Niki ESP     22000000  8.594441e+09   
314                 ZAMZAM ELECTRONICS TRADING     21700000  8.507819e+09   
336                          Vlad and Niki IDN     21000000  9.789122e+09   
401                             Manoj  parihar     19300000  2.642281e+08   
421                               Busy Fun Ltd     18800000  7.762077e+09   
428                       Smile Family Spanish     18600000  7.008250e+09   
468                           ýýýýýýýýý KIMPRO     17700000  1.920670e+10   
469                           Jane ASMR ��     17700000  7.387622e+09   
524                                Ivana Alawi     16600000  1.318443e+09   
562                          Vlad and Niki ARA     16100000  7.399296e+09   
567                               Panda Shorts     16100000  7.155736e+09   
585                                  MY FAMILY     15700000  6.153496e+09   
674                                 Filaretiki     14800000  1.578821e+10   
699                                Lotus Music     14500000  1.739129e+09   
703                Eli Kids - Cartoons & Songs     14500000  4.821183e+09   
723                          Vlad and Niki PRT     14400000  6.543282e+09   
769                   ýýýýýýýý ýý ýýýýýýýýýýýý     13900000  8.451755e+09   
791                              Zeinab Harake     13600000  8.968914e+08   
808                           Vlad y Niki Show     13400000  4.622629e+09   
811                                   Vibhu 96     13400000  2.056338e+07   
836                                     Vilmei     13200000  4.205665e+09   
845                                 WiederDude     13100000  2.733683e+09   
864                             Wolfoo Channel     13000000  3.015478e+08   
887                             FACT FIRE KING     12700000  1.159290e+09   
891                         Diana and Roma IND     12700000  5.567832e+09   
898                             Heidi y Zidane     12700000  6.001189e+09   
907                              Wolfoo Family     12500000  1.612540e+08   
913     Bebefinn - Nursery Rhymes & Kids Songs     12500000  4.163639e+09   
927                               DEV Ke Facts     12400000  1.971226e+09   
946                                      Panda     12300000  2.214684e+09   

    category  uploads               Country   channel_type  \
85       NaN      515         United States  Entertainment   
124      NaN     2738                Turkey          Games   
146      NaN      578         United States  Entertainment   
152      NaN      532         United States  Entertainment   
201      NaN      775         United States  Entertainment   
239      NaN      469  United Arab Emirates         People   
240      NaN       36         United States      Education   
245      NaN      590         United States  Entertainment   
294      NaN      526         United States  Entertainment   
301      NaN      515         United States  Entertainment   
314      NaN    15126  United Arab Emirates         People   
336      NaN      498         United States  Entertainment   
401      NaN      335                 India          Howto   
421      NaN        2                   NaN            NaN   
428      NaN      457      

     We will now check for possible values we can use to replace the null values within the 'category' column based on the 'channel_type' column.
     - This is a viable way of handling the null values as both variables are similar in nature.

In [741]:
# Displaying the 'category' and 'channel_type' values for cleaning
column1 = 'category'
column2 = 'channel_type'

categoryandchannel_df = youtube_df.dropna(subset=[column1, column2])

unique_values_column1 = categoryandchannel_df[column1].unique().tolist()
unique_values_column2 = categoryandchannel_df[column2].unique().tolist()

df_unique_values_column1 = pd.DataFrame(unique_values_column1, columns=[column1])
df_unique_values_column2 = pd.DataFrame(unique_values_column2, columns=[column2])

# Concatenate DataFrames horizontally
result_df = pd.concat([df_unique_values_column1, df_unique_values_column2], axis=1)

# Display the unique values side by side as DataFrames
result_df


category   channel_type
0                   Music          Music
1           Entertainment  Entertainment
2               Education      Education
3                   Shows         People
4          People & Blogs         Sports
5                  Gaming           Film
6                  Sports           News
7        Film & Animation          Games
8           Howto & Style         Comedy
9         News & Politics          Howto
10                 Comedy      Nonprofit
11               Trailers          Autos
12  Nonprofits & Activism           Tech
13   Science & Technology        Animals
14                 Movies            NaN
15         Pets & Animals            NaN
16       Autos & Vehicles            NaN
17        Travel & Events            NaN

     After going through the unique values, we have prepared the following values to be used as replacement based on the channel_type.
| channel_type | category |
| ----------- | ----------- |
| Games      | Gaming       |
| People   | People & Blogs        |
| Entertainment      | Entertainment       |
| Sports   | Sports        |
| Film      | Film & Animation       |
| Howto   | Howto & Style        |
| Education      | Education       |
| Tech   | Science & Technology        |
| Music   | Music    |

     For the remaining rows with neither "column" nor "channel_type" values, they will be removed.

In [742]:
# Replace category null values with closest value in channel_type

# Use the .loc method to set the new value based on the condition (where 'category' is null)
youtube_df.loc[(youtube_df['channel_type'] == 'Games') & (youtube_df['category'].isnull()), 'category'] = 'Gaming'
youtube_df.loc[(youtube_df['channel_type'] == 'People') & (youtube_df['category'].isnull()), 'category'] = 'People & Blogs'
youtube_df.loc[(youtube_df['channel_type'] == 'Entertainment') & (youtube_df['category'].isnull()), 'category'] = 'Entertainment'
youtube_df.loc[(youtube_df['channel_type'] == 'Sports') & (youtube_df['category'].isnull()), 'category'] = 'Sports'
youtube_df.loc[(youtube_df['channel_type'] == 'Film') & (youtube_df['category'].isnull()), 'category'] = 'Film & Animation'
youtube_df.loc[(youtube_df['channel_type'] == 'Howto') & (youtube_df['category'].isnull()), 'category'] = 'Howto & Style'
youtube_df.loc[(youtube_df['channel_type'] == 'Education') & (youtube_df['category'].isnull()), 'category'] = 'Education'
youtube_df.loc[(youtube_df['channel_type'] == 'Tech') & (youtube_df['category'].isnull()), 'category'] = 'Science & Technology'
youtube_df.loc[(youtube_df['channel_type'] == 'Music') & (youtube_df['category'].isnull()), 'category'] = 'Music'

# Drop remaining rows with null values for 'category' column
youtube_df.dropna(subset=['category'], inplace=True)

# Display the modified DataFrame
youtube_df

Youtuber  subscribers   video views        category  \
0                      T-Series    245000000  2.280000e+11           Music   
1                       MrBeast    166000000  2.836884e+10   Entertainment   
2    Cocomelon - Nursery Rhymes    162000000  1.640000e+11       Education   
3                     SET India    159000000  1.480000e+11           Shows   
4           ýýý Kids Diana Show    112000000  9.324704e+10  People & Blogs   
..                          ...          ...           ...             ...   
944               Natan por Aï¿     12300000  9.029610e+09          Sports   
945    Free Fire India Official     12300000  1.674410e+09  People & Blogs   
946                       Panda     12300000  2.214684e+09          Gaming   
947                 RobTopGames     12300000  3.741235e+08          Gaming   
948                Make Joke Of     12300000  2.129774e+09          Comedy   

     uploads         Country   channel_type  lowest_monthly_earnings  \
0      20082           India          Music                 564600.0   
1        741   United States  Entertainment                 337000.0   
2        966   United States      Education                 493800.0   
3     116536           India  Entertainment                 455900.0   
4       1111   United States  Entertainment                 182900.0   
..       ...             ...            ...                      ...   
944     1200          Brazil  Entertainment                 138100.0   
945     1500           India          Games                  16200.0   
946     2452  United Kingdom          Games                     17.0   
947       39          Sweden          Games                    968.0   
948       62           India         Comedy                   6000.0   

     highest_monthly_earnings  lowest_yearly_earnings  \
0                   9000000.0               6800000.0   
1                   5400000.0               4000000.0   
2                   7900000.0               5900000.0   
3                   7300000.0               5500000.0   
4                   2900000.0               2200000.0   
..                        ...                     ...   
944                 2200000.0               1700000.0   
945                  258900.0                194200.0   
946                     268.0                   201.0   
947                   15500.0                 11600.0   
948                   96000.0                 72000.0   

     highest_yearly_earnings  created_year    Population  
0                108400000.0        2006.0  1.366418e+09  
1                 64700000.0        2012.0  3.282395e+08  
2                 94800000.0        2006.0  3.282395e+08  
3                 87500000.0        2006.0  1.366418e+09  
4                 35100000.0        2015.0  3.282395e+08  
..                       ...           ...           ...  
944               26500000.0        2017.0  2.125594e+08  
945                3100000.0        2018.0  1.366418e+09  
946                   3200.0        2006.0  6.683440e+07  
947                 185800.0        2012.0  1.028545e+07  
948                1200000.0        2017.0  1.366418e+09  

[948 rows x 13 columns]

     We will also drop the "channel_type" column since we will no longer be using it.

In [743]:
# Drop 'channel_type' column
youtube_df.drop(columns='channel_type', inplace=True)

youtube_df

Youtuber  subscribers   video views        category  \
0                      T-Series    245000000  2.280000e+11           Music   
1                       MrBeast    166000000  2.836884e+10   Entertainment   
2    Cocomelon - Nursery Rhymes    162000000  1.640000e+11       Education   
3                     SET India    159000000  1.480000e+11           Shows   
4           ýýý Kids Diana Show    112000000  9.324704e+10  People & Blogs   
..                          ...          ...           ...             ...   
944               Natan por Aï¿     12300000  9.029610e+09          Sports   
945    Free Fire India Official     12300000  1.674410e+09  People & Blogs   
946                       Panda     12300000  2.214684e+09          Gaming   
947                 RobTopGames     12300000  3.741235e+08          Gaming   
948                Make Joke Of     12300000  2.129774e+09          Comedy   

     uploads         Country  lowest_monthly_earnings  \
0      20082           India                 564600.0   
1        741   United States                 337000.0   
2        966   United States                 493800.0   
3     116536           India                 455900.0   
4       1111   United States                 182900.0   
..       ...             ...                      ...   
944     1200          Brazil                 138100.0   
945     1500           India                  16200.0   
946     2452  United Kingdom                     17.0   
947       39          Sweden                    968.0   
948       62           India                   6000.0   

     highest_monthly_earnings  lowest_yearly_earnings  \
0                   9000000.0               6800000.0   
1                   5400000.0               4000000.0   
2                   7900000.0               5900000.0   
3                   7300000.0               5500000.0   
4                   2900000.0               2200000.0   
..                        ...                     ...   
944                 2200000.0               1700000.0   
945                  258900.0                194200.0   
946                     268.0                   201.0   
947                   15500.0                 11600.0   
948                   96000.0                 72000.0   

     highest_yearly_earnings  created_year    Population  
0                108400000.0        2006.0  1.366418e+09  
1                 64700000.0        2012.0  3.282395e+08  
2                 94800000.0        2006.0  3.282395e+08  
3                 87500000.0        2006.0  1.366418e+09  
4                 35100000.0        2015.0  3.282395e+08  
..                       ...           ...           ...  
944               26500000.0        2017.0  2.125594e+08  
945                3100000.0        2018.0  1.366418e+09  
946                   3200.0        2006.0  6.683440e+07  
947                 185800.0        2012.0  1.028545e+07  
948                1200000.0        2017.0  1.366418e+09  

[948 rows x 12 columns]

In [744]:
# Names of the columns that will be modified
columns_to_modify = ['subscribers', 'video views', 'Population']

# Define a function to convert values to millions and change data type
def convert_to_millions_and_int(value):
    return (float(value) / 1e6) if pd.notna(value) else value

# Apply the conversion function to the specified columns
for column in columns_to_modify:
    youtube_df[column] = youtube_df[column].apply(convert_to_millions_and_int)

# Rename columns to indicate values are in millions
youtube_df.rename(columns={'subscribers': 'subscribers (in millions)', 'video views': 'video views (in millions)', 'Population': 'Population (in millions)'}, inplace=True)

youtube_df

Youtuber  subscribers (in millions)  \
0                      T-Series                      245.0   
1                       MrBeast                      166.0   
2    Cocomelon - Nursery Rhymes                      162.0   
3                     SET India                      159.0   
4           ýýý Kids Diana Show                      112.0   
..                          ...                        ...   
944               Natan por Aï¿                       12.3   
945    Free Fire India Official                       12.3   
946                       Panda                       12.3   
947                 RobTopGames                       12.3   
948                Make Joke Of                       12.3   

     video views (in millions)        category  uploads         Country  \
0                228000.000000           Music    20082           India   
1                 28368.841870   Entertainment      741   United States   
2                164000.000000       Education      966   United States   
3                148000.000000           Shows   116536           India   
4                 93247.040539  People & Blogs     1111   United States   
..                         ...             ...      ...             ...   
944                9029.609749          Sports     1200          Brazil   
945                1674.409945  People & Blogs     1500           India   
946                2214.684303          Gaming     2452  United Kingdom   
947                 374.123483          Gaming       39          Sweden   
948                2129.773714          Comedy       62           India   

     lowest_monthly_earnings  highest_monthly_earnings  \
0                   564600.0                 9000000.0   
1                   337000.0                 5400000.0   
2                   493800.0                 7900000.0   
3                   455900.0                 7300000.0   
4                   182900.0                 2900000.0   
..                       ...                       ...   
944                 138100.0                 2200000.0   
945                  16200.0                  258900.0   
946                     17.0                     268.0   
947                    968.0                   15500.0   
948                   6000.0                   96000.0   

     lowest_yearly_earnings  highest_yearly_earnings  created_year  \
0                 6800000.0              108400000.0        2006.0   
1                 4000000.0               64700000.0        2012.0   
2                 5900000.0               94800000.0        2006.0   
3                 5500000.0               87500000.0        2006.0   
4                 2200000.0               35100000.0        2015.0   
..                      ...                      ...           ...   
944               1700000.0               26500000.0        2017.0   
945                194200.0                3100000.0        2018.0   
946                   201.0                   3200.0        2006.0   
947                 11600.0                 185800.0        2012.0   
948                 72000.0                1200000.0        2017.0   

     Population (in millions)  
0                 1366.417754  
1                  328.239523  
2                  328.239523  
3                 1366.417754  
4                  328.239523  
..                        ...  
944                212.559417  
945               1366.417754  
946                 66.834405  
947                 10.285453  
948               1366.417754  

[948 rows x 12 columns]

### The "category" variable

     We want to check out 
     

In [747]:
# insert code here
youtube_df['category'].unique()

array(['Music', 'Entertainment', 'Education', 'Shows', 'People & Blogs',
       'Gaming', 'Sports', 'Film & Animation', 'Howto & Style',
       'News & Politics', 'Comedy', 'Trailers', 'Nonprofits & Activism',
       'Science & Technology', 'Movies', 'Pets & Animals',
       'Autos & Vehicles', 'Travel & Events'], dtype=object)

In [748]:
null_values = youtube_df[youtube_df["channel_type"].isnull()]
null_values

KeyError: 'channel_type'

### The "Monthly Earnings" variable

     We want to check out 
     

In [ ]:
# insert code here

### The "Highest Monthly Earnings" variable
 We want to check out 
 

In [ ]:
# insert code here

### The "Subscribers" variable

In [ ]:
# insert code here

### The "video views" variable



In [ ]:
# insert code here

### The "uploads" variable
 We want to check out 
 

In [ ]:
# insert code here

# Exploratory Data Analysis

    Placeholder detailsl

### What are the Highest Earning Categories?

    LMAO Try ko lang
    

In [ ]:
# insert Code Here

### Are there any correlation between views and monthly earnings?

    Placeholder details


### Are there any top Youtube channels with a lot of subscribers but has no video uploads?

    To find bot channels in the dataset
    

### Are there any top Youtube Channels with less than 2 videos and only a little bit of views and uploads?
    
    To extend the exploration on finding bots channel in the dataset_____
    

### Are there any correlations between views and subscribers?

 Placeholder details